datalad/downloaders/providers.py

```diff
@@ -337,6 +367,119 @@ def get_provider(self, url, only_nondefault=False):
                   scheme, provider)
         return provider
 
+    def enter_new(self, url=None, auth_types=[]):
+        from datalad.ui import ui
+        name = None
+        if url:
+            ri = RI(url)
+            for f in ('hostname', 'name'):
+                try:
+                    # might need sanitarization
+                    name = str(getattr(ri, f))
+                except AttributeError:
+                    pass
+        known_providers_by_name = {p.name: p for p in self._providers}
+        PROVIDERS_USER_DIR = self._get_providers_dirs()['user']
```

Comment:
This is local to this function.  Why the allcaps?

Reply:
I finally found CAPSLOCK useful! ;-)  I guess leftover from when I hoped to have those just a process-wide constants... will lowercase